batch/batch/front_end/front_end.py

```diff
@@ -227,17 +240,17 @@ async def _handle_ui_error(session, f, *args, **kwargs):
         raise
 
 
-async def _handle_api_error(f, *args, **kwargs):
+async def _handle_api_error(f: Callable[P, Awaitable[T]], *args: P.args, **kwargs: P.kwargs) -> Optional[T]:
```

Comment:
This should be a separate PR, but these new types and all those `assert ... is not None` reveal another issue.

This function conflates two distinct pieces of functionality:
1. Make functions returning no values idempotent
2. Convert `BatchUserError` into an HTTPResponse.

I'm a bit unsure if BatchOperationAlreadyCompletedError is the right thing to use. It seems a bit confusing if I am at the delete billing project form, then press submit, then I lose my WiFi, then refresh the page (clicking OK when it says "do you want to resubmit this form"), then I get "this billing project already is deleted". I guess that tells me that the first request actually did make it through, but as a user, all that matters is that the BP doesn't exist anymore, right? If I mistype the name I'd get a 404, so I can't accidentally think I deleted a different BP.

Regardless, I think we should have `_handle_api_error` which lacks the first except and `_handle_idempotence_and_api_error` which calls `_handle_api_error` wrapped in a try-except.